src/gtest/test_transaction_builder.cpp

```diff
@@ -69,14 +67,13 @@ TEST(TransactionBuilder, Invoke)
     // Create a Sapling-only transaction
     // 0.0004 z-ZEC in, 0.00025 z-ZEC out, 0.0001 t-ZEC fee, 0.00005 z-ZEC change
     auto builder2 = TransactionBuilder(consensusParams, 2);
-    ASSERT_TRUE(builder2.AddSaplingSpend(expsk, note, anchor, witness));
+    builder2.AddSaplingSpend(expsk, note, anchor, witness);
```

Comment:
This is because we now throw an exception when this would have previously returned false. Either way the test should fail here if we could not add the sapling spend.

Reply:
are you 100% sure? I see an exception thrown in `AddSaplingSpend` in one case, but then two more commands

`spends.emplace_back(expsk, note, anchor, witness);`

     `mtx.valueBalance += note.value();`		     

that afaik don't throw an exception on failue (or do they?)

Reply:
I am not sure how either `spends.emplace_back(expsk, note, anchor, witness);` or `mtx.valueBalance += note.value();` could fail, but once we had gotten to that point in the method we were either going to `return true;`, or throw an exception on one of those lines. If they threw an exception previously, they will continue to do so. If there were additional failure cases in this method (which I am not sure that there are), they were not being checked for.

Reply:
I think you convinced me:)